python/cudf/cudf/_lib/column.pyx

```diff
@@ -386,14 +387,19 @@ cdef class Column:
             tid = libcudf_types.type_id.LIST
         elif is_struct_dtype(self.dtype):
             tid = libcudf_types.type_id.STRUCT
+        elif is_decimal_dtype(self.dtype):
+            tid = libcudf_types.type_id.DECIMAL64
         else:
             tid = <libcudf_types.type_id> (
                 <underlying_type_t_type_id> (
                     np_to_cudf_types[np.dtype(data_dtype)]
                 )
             )
-
-        cdef libcudf_types.data_type dtype = libcudf_types.data_type(tid)
+        cdef libcudf_types.data_type dtype = (
+            libcudf_types.data_type(tid, -self.dtype.scale)
+            if tid == libcudf_types.type_id.DECIMAL64
```

Comment:
What about DECIMAL32?

Reply:
I thought one of the design decisions we made before this PR was to support just DECIMAL64 for now. Do we want both?

Reply:
I guess I have a fear that some IO somewhere is going to be read into a `DECIMAL32` and we don't have the ability to upcast or anything for now, but otherwise we should be fine.

Reply:
Should we explicitly throw for now on getting `DECIMAL32` to make it clear?

Reply:
@shwina I think this is the only item that should still be discussed / addressed before we merge